src/lib/evloop/workqueue.c

```diff
@@ -85,6 +85,9 @@ struct threadpool_s {
   void *(*new_thread_state_fn)(void*);
   void (*free_thread_state_fn)(void*);
   void *new_thread_state_arg;
+
+  /** Function to start a thread. Should return a negative number on error. */
+  int (*thread_spawn_fn)(void (*func)(void *), void *data);
```

Comment:
I think it might be convenient to have a typedef for this function pointer type; the documentation should also explain what the arguments mean.